scalding-serialization/src/main/scala/com/twitter/scalding/serialization/OrderedSerialization.scala

```diff
@@ -101,6 +101,30 @@ object OrderedSerialization {
     case NonFatal(e) => CompareFailure(e)
   }
 
+  def viaTransform[T, U](
+    packFn: T => U,
+    unpackFn: U => T)(implicit otherOrdSer: OrderedSerialization[U]): OrderedSerialization[T] =
+    new OrderedSerialization[T] {
+
+      override def hash(t: T) = otherOrdSer.hash(packFn(t))
```

Comment:
I think it is safe: we do one read (which may be old). Then that one read has linked two values: `(T, U)`. If the `T` matches, you're safe to use the `U` no matter what.

If you are out of date, the worst case is you recompute needlessly, right?

You're right about the `equiv`. Mistake there. That compares on `U` obviously. We could use `==` here and assume that if `a == b` that implies `equiv(a, b)`, seems sane (that's not saying they are the same, just that if equals returns true, then equiv would as well. Can't think of a good counter example to that).